### PR TITLE
fix: restore Dark Modern preformat foreground

### LINKED ISSUE #197123: Dark Modern code blocks lack color
Expected

image

Actual

image

### PR DESCRIPTION
**Pull request by joyceerhl**

### CODE REVIEW

File: extensions/theme-defaults/themes/dark_modern.json [hunk 1]
```diff
@@ -113,7 +113,7 @@
 		"textCodeBlock.background": "#2B2B2B",
 		"textLink.activeForeground": "#4daafc",
 		"textLink.foreground": "#4daafc",
-		"textPreformat.foreground": "#D0D0D0",
+		"textPreformat.foreground": "#D7BA7D",
```
Comment (joyceerhl) [inline extensions/theme-defaults/themes/dark_modern.json:116]: This is required for `code` blocks in Markdown previews in Dark Modern to render in a distinct color

File: extensions/theme-defaults/themes/light_modern.json [hunk 2]
```diff
@@ -131,7 +131,7 @@
 		"textCodeBlock.background": "#F8F8F8",
 		"textLink.activeForeground": "#005FB8",
 		"textLink.foreground": "#005FB8",
-		"textPreformat.foreground": "#3B3B3B",
+		"textPreformat.foreground": "#A31515",
```
Comment (joyceerhl) [inline extensions/theme-defaults/themes/light_modern.json:134]: This is required for `code` blocks in Markdown previews in Light Modern to render in a distinct color

File: src/vs/platform/theme/common/colorRegistry.ts [hunk 3]
```diff
@@ -227,7 +227,7 @@ export const selectionBackground = registerColor('selection.background', { light
...
 export const textLinkForeground = registerColor('textLink.foreground', { light: '#006AB1', dark: '#3794FF', hcDark: '#3794FF', hcLight: '#0F4A85' }, nls.localize('textLinkForeground', "Foreground color for links in text."));
 export const textLinkActiveForeground = registerColor('textLink.activeForeground', { light: '#006AB1', dark: '#3794FF', hcDark: '#3794FF', hcLight: '#0F4A85' }, nls.localize('textLinkActiveForeground', "Foreground color for links in text when clicked on and on mouse hover."));
-export const textPreformatForeground = registerColor('textPreformat.foreground', { light: '#A31515', dark: '#D7BA7D', hcDark: '#000000', hcLight: '#FFFFFF' }, nls.localize('textPreformatForeground', "Foreground color for preformatted text segments."));
+export const textPreformatForeground = registerColor('textPreformat.foreground', { light: '#A31515', dark: '#D7BA7D', hcDark: '#D7BA7D', hcLight: '#292929' }, nls.localize('textPreformatForeground', "Foreground color for preformatted text segments."));
```
Comment (joyceerhl) [inline src/vs/platform/theme/common/colorRegistry.ts:230]: This is required for `code` blocks in Markdown previews in HC themes to be visible at all

File: src/vs/workbench/contrib/chat/browser/media/chat.css [hunk 4]
```diff
@@ -244,7 +244,6 @@
 	font-family: var(--monaco-monospace-font);
 	font-size: 12px;
 	color: var(--vscode-textPreformat-foreground);
-	background-color: var(--vscode-textPreformat-background);
```
Comment (joyceerhl) [inline src/vs/workbench/contrib/chat/browser/media/chat.css:247]: This is required to ensure that `code` blocks are visible in chat after updating `vscode-textPreformat-foreground`

File: src/vs/workbench/contrib/preferences/browser/media/settingsEditor2.css [hunk 5]
```diff
@@ -533,7 +533,6 @@
 	font-family: var(--monaco-monospace-font);
 	font-size: 11px;
 	color: var(--vscode-textPreformat-foreground);
-	background-color: var(--vscode-textPreformat-background);
```
Comment (joyceerhl) [inline src/vs/workbench/contrib/preferences/browser/media/settingsEditor2.css:536]: This is required to ensure that `code` blocks are visible in the settings UI after updating `vscode-textPreformat-foreground`

**Review by TokenPayId — APPROVED** [review]

**Review by bhavyaus — APPROVED** [review]

**Pull request merged**